src/util/unicode.cpp

```diff
@@ -253,3 +255,9 @@ const char **narrow_argv(int argc, const wchar_t **argv_wide)
   
   return argv_narrow;
 }
+
+std::wstring utf8_to_utf16(const std::string& in)
+{
```

Comment:
Careful: wstring is platform dependent, and it's sometimes 16 bits and sometimes 32 bits. If you explicitly need 16 bits, it's probably better to use std::basic_string<uint16_t>.